src/python/pants/source/payload_fields.py

```diff
@@ -35,14 +34,6 @@ def __init__(self, sources: FilesetWithSpec, ref_address=None) -> None:
         self._sources = self._validate_sources(sources)
         self._ref_address = ref_address
 
-    @property
-    def source_root(self):
```

Comment:
Could we use the same code in `target.py` to deprecate this instead of outright removal? While technically not API public, a deprecation is more friendly to plugin developers.